transform-plugins/docs/Run-transform.md

```diff
@@ -0,0 +1,92 @@
+# Run Transform
+
+
+Description
+-----------
+Runs an executable binary which is installed and available on the local filesystem of the Hadoop nodes. Run transform
+plugin allows the user to read the structured record as input and returns the output record, to be further processed
+downstream in the pipeline.
+
+Use Case
+--------
+In enterprise, there are some existing tools or executable binaries that perform complex transformations of data.
+This plugin can be used, when user would like to execute these type of binaries that will read the structured record
+as input, process it and retrieve the results back to the pipeline.
+
+Properties
+----------
+**commandToExecute:** Command that will contain the full path to the executable binary present on the local
+filesystem of the Hadoop nodes as well as how to execute that binary. It should not contain any input arguments. For
+example, java -jar /home/user/ExampleRunner.jar, if the binary to be executed is of type jar.
+
+**fieldsToProcess:** A comma-separated sequence of the fields to be used as variable input arguments for binary to be
+executed. For example, 'firstname' or 'firstname,lastname' in case of multiple arguments. Please make sure that the
+sequence of fields/arguments is in the order as expected by binary. (Macro Enabled)
+
+**fixedArguments:** A space-separated sequence of the fixed input arguments that will be passed to the executable
+binary. Please make sure that the sequence of arguments is in the order as expected by binary. All the fixed arguments
+will be followed by the variable arguments, provided through 'Fields to Process for Variable Arguments'. (Macro enabled)
+
+**outputField:** The field name that holds the output of the executable binary.
+
+**outputFieldType:** Schema type of the 'Output Field'. Supported types are: boolean, bytes, double, float, int, long
+and string.
+
+Conditions
+----------
+Executable binary and its dependencies must be available on all the nodes of the Hadoop cluster, prior to the execution
+of binary.
+
+Executable binary will always read the input through STDIN and must generate the STDOUT for each input record. Also,
+errors emitted by the executable through STDERR will be captured in logs.
+
+Executable binary can take 0 to N arguments. Source for the variable arguments will always be the structured records
+coming through the Hydrator source stage. Required fields can be provided using 'fieldsToProcess' property.
+
+Fixed input arguments(if any), will always be followed by the variable input arguments. This will be the format of the
+arguments that will be passed to the executable binary.
+
+Arguments should be in the expected order and the supported format. Any mismatch in the sequence of the arguments will
+result into the runtime failure.
+
+Supported types for binary to be executed are: 'exe, sh, bat and jar'.
```

Comment:
We had implemented considering the design that we will support these four types. "exe, sh, bat and jar".
However, I tested with python script, currently, its failing.
I will look into this and will update accordingly.